.github/workflows/pull-request-check.yaml

```
@@ -0,0 +1,46 @@
+name: Pull Request Check
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  lint:
+    name: Run lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22.x'
+      - run: npm ci
+      - run: npm run lint
+
+  test:
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22.x'
+      - run: npm ci
+      - run: npm run test
```

```suggestion
      - name: Setup
        uses: ./.github/actions/setup
        
      - name: Run Test
         run: npm run test        
```

node.jsのsetupから依存関係のインストールまでは各jobで共通なので、composite actionで定義し、各jobはcomposite actionを呼び出す感じにしたいです。

---

composite actionの方では、以下に対してキャッシュを利用するようにして高速化を図りたいです
- actions/setup-node
- actions/cache

参考: [GitHub Action で npm ci 高速化のために cache を使ってみる実験ノート #Node.js - Qiita](https://qiita.com/aKuad/items/4a12ddb6326683155dfd)